2024 카카오 윈터 인턴십/PGS_가장많이받은선물_홍예진.java

```diff
@@ -0,0 +1,93 @@
+import java.util.*;
+
+class Solution {
+    int n;
+    Map<String, Friend> map;
+    Friend[] friendArr;
+    int[][] giveCount;
+    public void init(String[] friends, String[] gifts){
+        n = friends.length;
+        map = new HashMap<>();
+        friendArr = new Friend[n];
+        giveCount = new int[n][n];
+
+        for(int number = 0; number < n; number++){
+            String friendName = friends[number];
+            friendArr[number] = new Friend(friendName, number, 0);
+            map.put(friendName, friendArr[number]);
+        }
+
+        for(String gift : gifts){
+            StringTokenizer st = new StringTokenizer(gift);
+            String giverName = st.nextToken();
+            String receiverName = st.nextToken();
+
+            Friend giver = map.get(giverName);
+            Friend receiver = map.get(receiverName);
+
+            giver.giftIndex++;
+            receiver.giftIndex--;
+            giveCount[giver.number][receiver.number]++;
+        }
+
+    }
+    public int findReceiver(int i, int j){
+        int diff = giveCount[i][j] - giveCount[j][i]; // 선물을 준 횟수의 차이
+        if(diff > 0){
+            return i;
+        } else if(diff < 0){
+            return j;
+        }
+
+        Friend fi = friendArr[i];
+        Friend fj = friendArr[j];
+        diff = fi.giftIndex - fj.giftIndex;
+        if(diff > 0){
+            return i;
+        } else if(diff < 0){
+            return j;
+        }
+
+        // 두 사람의 선물 지수가 같을 때
+        return -1;
+    }
+
+    public int getMax(int[] nextGiftCount){
+        int maxIndex = 0;
+        int max = nextGiftCount[maxIndex];
+        for(int i = 0; i < n; i++){
+            if(max >= nextGiftCount[i]) continue;
+            maxIndex = i;
+            max = nextGiftCount[i];
+        }
+        return max;
+    }
```

Comment:
```suggestion
    public int getMax(int[] nextGiftCount){
        int result = 0;
        
        for(int i = 0; i < n; i++){
            result = Math.max(result, nextGiftCount[i]);
        }
        
        return result;
    }
```

누가 선물을 가장 많이 받는지가 아니라, 선물을 가장 많이 받은 사람이 몇 개를 받는지를 리턴하는 거니까 `maxIndex`는 없어도 될것같아요.